moon/rospy_rover.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+"""
+  Wait for all necessary ROS sensors
+"""
+import time
+import struct
+import math
+from io import BytesIO
+from threading import RLock
+
+import zmq
+
+import rospy
+from std_msgs.msg import *  # Float64, JointState
+from sensor_msgs.msg import *
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Point
+
+# SRCP2 specific
+from srcp2_msgs.msg import Qual1ScoringMsg, VolSensorMsg, Qual3ScoringMsg
+from srcp2_msgs.srv import (ToggleLightSrv, LocalizationSrv, Qual1ScoreSrv,
+                            AprioriLocationSrv, HomeLocationSrv, HomeAlignedSrv)
+
+
+g_socket = None
+g_lock = RLock()
+
+
+def socket_send(data):
+    global g_socket, g_lock
+    assert g_socket is not None
+    with g_lock:
+        g_socket.send(data)
+
+
+def callback_lidar(data):
```

Comment:
later I would like to remove both `callback_lidar` and `callback_imu` and use `callback_topic` only ... but it requires more changes in `rosmsg.py` so I plan it in next iteration (not in this PR)